File: src/main/java/org/evertones/persistence/modules/client/ClientDetailsRepository.java
package org.evertones.persistence.modules.client;

import org.evertones.model.modules.client.Client;
import org.evertones.model.modules.client.ClientDetails;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QueryDslPredicateExecutor;
import org.springframework.stereotype.Repository;

@Repository
public interface ClientDetailsRepository
        extends JpaRepository<ClientDetails, Integer>, QueryDslPredicateExecutor<ClientDetails> {
}

File: src/main/scala/org/evertones/persistence/modules/client/ClientDetailsRepository.scala
package org.evertones.persistence.modules.client

import org.evertones.model.modules.client.ClientDetails
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QueryDslPredicateExecutor
import org.springframework.stereotype.Repository

@Repository
trait ClientDetailsRepository extends JpaRepository[ClientDetails, Integer]
    with QueryDslPredicateExecutor[ClientDetails] {
}
